### ISSUE TITLE
Http package fails to read an URL

### ISSUE THREAD

**Issue by DartBot** [dart-lang/http#9]
<a href="https://github.com/kaisellgren"><img src="https://avatars.githubusercontent.com/u/148256?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [kaisellgren](https://github.com/kaisellgren)**
_Originally opened as dart-lang/sdk#9675_

----

This code using the HTTP package fails:

&nbsp;&nbsp;&nbsp;&nbsp;http.read('https://www.chms.edu.mx/')

The result is:

Uncaught Error: SocketIOException: RawSecureSocket error (Unexpected handshake error in client) (OS Error: errno = -8181)
Unhandled exception:
SocketIOException: RawSecureSocket error (Unexpected handshake error in client) (OS Error: errno = -8181)
#­0      \_FutureImpl._scheduleUnhandledError.&lt;anonymous closure&gt; (dart:async:347:9)
#­1      Timer.Timer.&lt;anonymous closure&gt; (dart:async-patch:11:15)
#­2      \_Timer._createTimerHandler._handleTimeout (dart:io:6031:28)
#­3      \_Timer._createTimerHandler._handleTimeout (dart:io:6039:7)
#­4      \_Timer._createTimerHandler.&lt;anonymous closure&gt; (dart:io:6047:23)
#­5      \_ReceivePortImpl._handleMessage (dart:isolate-patch:40:92)

**Comment by DartBot** [comment]
<a href="https://github.com/sgjesse"><img src="https://avatars.githubusercontent.com/u/1343914?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sgjesse](https://github.com/sgjesse)**

----

We are missing turning the NSS error codes into test. You can see the error codes on http://www.mozilla.org/projects/security/pki/nss/ref/ssl/sslerr.html.

The error -8181 indicate an expired certificate.

Currently there are no way to bypass a bad certificate using either package:http or dart:io directly. The SecureSocket and RawSecureSocket has onBadCertificate callbacks thich can be used to bypadd invalid certificates.
______
*Removed [Type-Defect](../labels/Type-Defect) label.*
*Added [Type-Enhancement](../labels/Type-Enhancement), [Area-Pkg](../labels/Area-Pkg), [Triaged](../labels/Triaged) labels.*
*Marked this as being blocked by dart-lang/sdk#9689.*

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

*Added [Pkg-Http](../labels/Pkg-Http) label.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

This is out of scope for direct support in the http package, but it can be accomplished relatively easily by passing in an HttpClient with an onBadCertificate callback to [new IOClient].
______
*Added [NotPlanned](../labels/NotPlanned) label.*
